src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,250 @@
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  // Extracting the filename
+  const std::string filename = CLI::GetParam<std::string>("actual_dataset");
+  // This is very dangerous, Let's add a check tommorrow.
+  std::string column_delimiter;
+  if (CLI::HasParam("column_delimiter"))
+  {
+    column_delimiter = CLI::GetParam<std::string>("column_delimiter");
+    // Allow only 3 delimeter.
+    RequireParamValue<std::string>("column_delimiter", [](std::string del)
+        { return del == "\t" || del == "," || del == " "; }, true,
+        "Delimiter should be either \\t (tab) or , (comma) or ' ' (space) ");
+  }
+  else
+  {
+      if (data::Extension(filename) == "csv")
+      {
+        column_delimiter = ",";
+        Log::Warn << "Found csv Extension, taking , as column_delimiter. \n";
+      }
+      else if (data::Extension(filename) == "tsv" ||
+          data::Extension(filename) == "txt")
+      {
+        std::cout<<"fdsf \n";
+        column_delimiter = "\t";
+        Log::Warn << "Found tsv or txt Extension, taking \\t as"
+        "column_delimiter. \n";
+      }
+      else
+      {
+        Log::Warn << "column_delimiter not specified, taking default value \n";
+      }
+  }
+  // Handling Dimension vector
+  std::vector<std::string> temp_dimension = CLI::GetParam<std::vector<
+    std::string> >("dimension");
+  std::vector<size_t>dimension;
+  int columnstartindex, columnendindex;
+  size_t found;
+  for (size_t i = 0; i < temp_dimension.size(); i++)
+  {
+    found = temp_dimension[i].find('-');
+    if ( found != string::npos)
+    {
+      // Has a range include, something of type a-b.
+      columnstartindex = std::stoi(temp_dimension[i].substr(0, found));
+      columnendindex = std::stoi(temp_dimension[i].substr(found+1,
+          temp_dimension[i].length()));
+      for (int i = columnstartindex; i <= columnendindex; i++)
+      {
+        dimension.push_back(i);
+      }
+    }
+    else
+    {
+      dimension.push_back(std::stoi(temp_dimension[i]));
+    }
+  }
+  std::cout<<"\n";
+  // Sorting neccessary
+  std::sort(dimension.begin(), dimension.end());
+  // Extracting the Contents of file
+  // File pointer
+  ifstream fin;
+  // Open an existing file
+  fin.open(filename);
+  if (!fin.is_open())
+  {
+    throw std::runtime_error("Unable to open input file");
+  }
+  std::string line, word;
+  std::vector<std::vector<std::string>> dataset;
+  boost::string_view token, copy;
+  stringstream streamLine;
+  while (std::getline(fin, line))
+  {
+    streamLine.str(line);
+    dataset.push_back(std::vector<std::string>());
+    // delimeter[0] becase the standard function accepts char as input.
+    while (std::getline(streamLine, word, column_delimiter[0]))
+    {
+      dataset[dataset.size()-1].push_back(word);
+    }
+    streamLine.str(std::string());
+  }
```

Comment:
@lozhnikov , Could you help me here, while reading the lines from a file, it is just reading the first line correctly into stringstream, for subsquent line it is not working, I added cout statements and saw that the contents are being added but it is not reading from stringstream 

![Screenshot from 2019-07-20 22-47-27](https://user-images.githubusercontent.com/21286134/61582064-a4598700-ab43-11e9-873c-a34d5dd12684.png)

check the output, I have added functionality for `3-5` and other as same, but it is just for one row.

Reply:
Ok, So issue is resolved, it is reading all rows :), The punctuation has been removed from all columns

![Screenshot from 2019-07-21 17-27-24](https://user-images.githubusercontent.com/21286134/61590980-d4556880-abdd-11e9-8d2b-c1835af21626.png)